test/framework/test_client_factory.cpp

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "framework/test_client_factory.hpp"
+
+#include "common/bind.hpp"
+#include "framework/test_logger.hpp"
+#include "logger/logger_manager.hpp"
+#include "network/impl/channel_factory.hpp"
+#include "network/impl/channel_pool.hpp"
+
+using namespace std::literals::chrono_literals;
+
+template <typename Collection, typename Elem>
+void remove_elem(Collection &collection, const Elem &elem) {
+  collection.erase(std::remove(collection.begin(), collection.end(), elem));
```

Comment:
How does this work? `std::remove` "Removes all elements satisfying specific criteria from the range [first, last) and returns a past-the-end iterator for the new end of the range. ". What is `collection.erase` call used for?